### ISSUE TITLE
Support for NumPy 2.0

### ISSUE THREAD

**Issue by vnmabus** [jdtuck/fdasrsf_python#56]
NumPy 2.0 has been recently released, and modules compiled against previous versions are incompatible with it (but compiling against 2.0 apparently IS compatible with previous versions). When using NumPy 2.0 with code depending on this module I got the following message:
"A module that was compiled using NumPy 1.x cannot be run in
NumPy 2.0.0 as it may crash. To support both 1.x and 2.x
versions of NumPy, modules must be compiled with NumPy 2.0.
Some module may need to rebuild instead e.g. with 'pybind11>=2.12'.

If you are a user of the module, the easiest solution will be to
downgrade to 'numpy<2' or try to upgrade the affected module.
We expect that some modules will need time to support NumPy 2.
"

Please, if you can compile against NumPy 2.0, that would be very helpful.

**Comment by jdtuck** [comment]
I cannot support this till `oldest-supported-numpy` is updated to support this for build. This will have to wait

**Comment by jdtuck** [comment]
looks like a few are updated, let me test a few things today

**Comment by jdtuck** [comment]
`oldest-supported-numpy` is no longer needed per https://github.com/scipy/oldest-supported-numpy. Will work on this hopefully today.

**Comment by jdtuck** [comment]
fixed in main and release to pypi pending on CI
